File: BACKEND/TransaccionService/src/main/java/dev/jvall/TransaccionService/domain/Transaccion.java
package dev.jvall.TransaccionService.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import reactor.util.annotation.Nullable;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Transaccion {
    private Integer idtransaccion;
    private Double monto;
    private String fecha;
    @Nullable
    private Integer idorigen;
    private String tipo;
    private String descripcion;
}

File: FRONTEND/sm-web/src/app/backend/interfaces/transaccion.ts
export interface Transaccion{
    idtransaccion:number,
    monto:number,
    fecha:string,
    idorigen?:number,
    tipo:string,
    descripcion:string
}
export interface TransaccionRequest{
    monto: number,
    fecha: string,
    idorigen?:number,
    idtipo: number,
    descripcion: string
}